### DISCUSSION TITLE
eth_estimateGas error

### DISCUSSION THREAD

**Discussion by pblaas** [bancorprotocol/fastlane-bot#187] [General]
Hi, 

I'm running the latest main branch of the fastlane bot.  Unfortunately the bot doesn't seem able to make any transactions. 
In Alchemy the eth_estimateGas error is showing up.

The error message in Alchemy look like this:
`{
jsonrpc:"2.0",
id:58,
error:{
code:-32000,
message:"err: max fee per gas less than block base fee: address 0x468d9CcA86531f8ca95D724bC43DE898d0a9d458, maxFeePerGas: 54383373754 baseFee: 57529359482 (supplied gas 1975741)"
}
}`

Some info from the bot logs:
```
 13160	2023-11-09 14:39:54,495 [fastlane:INFO] - Found 12 eligible arb opportunities.
 13161	2023-11-09 14:40:04,487 [fastlane:INFO] - [2023-11-09T14:40:04::1699540804] |calculated_arb| == {'type': 'multi', 'profit_bnt': 89.1177, 'profit_usd': 65.3268, 'flashloan': [{'token': 'USDC-eB48', 'amount': 4983.861, 'profit': 65.3268}], 'trades': [{'trade_index': 0, 'exchange': 'carbon_v1', 'tkn_in': 'USDC-eB48', 'amount_in': 4983.861, 'tkn_out': 'WETH-6Cc2', 'amt_out': 2.6068, 'cid0': '8841057382'}, {'trade_index': 1, 'exchange': 'uniswap_v2', 'tkn_in': 'WETH-6Cc2', 'amount_in': 2.6068, 'tkn_out': 'USDC-eB48', 'amt_out': 5049.1878, 'cid0': 'b8da7e6be2'}]}
 13162	2023-11-09 14:40:05,443 [fastlane:INFO] - Validating trade...
 13163	2023-11-09 14:40:06,730 [fastlane:INFO] - Failed to estimate gas due to exception local variable 'transaction' referenced before assignment, scrapping transaction :(.
```

Another example:
```
  2137	2023-11-09 09:40:29,411 [fastlane:INFO] - Found 1 eligible arb opportunities.
  2138	2023-11-09 09:40:30,317 [fastlane:INFO] - [2023-11-09T09:40:30::1699522830] |calculated_arb| == {'type': 'multi', 'profit_bnt': 61.6046, 'profit_usd': 43.5056, 'flashloan': [{'token': 'ETH-EEeE', 'amount': 0.0767, 'profit': 0.0227}], 'trades': [{'trade_index': 0, 'exchange': 'uniswap_v2', 'tkn_in': 'WETH-6Cc2', 'amount_in': 0.0767, 'tkn_out': 'CSCS-cFB7', 'amt_out': 6850.5315, 'cid0': 'e2e3d800ad'}, {'trade_index': 1, 'exchange': 'carbon_v1', 'tkn_in': 'CSCS-cFB7', 'amount_in': 6850.5315, 'tkn_out': 'WETH-6Cc2', 'amt_out': 0.0994, 'cid0': '8158873117'}]}
  2139	2023-11-09 09:40:31,954 [fastlane:INFO] - Validating trade...
  2140	2023-11-09 09:40:32,922 [fastlane:INFO] - Failed to estimate gas due to exception local variable 'transaction' referenced before assignment, scrapping transaction :(.
```

Not sure if I missed anything regarding the setup, is this expected behaviour due to insufficient funds on the connect eth wallet or something else. The wallet does not have 5K USD which might have any relevance with the first example from the logs?  The wallet does have 200 USD available currently which should have allowed me to make that second example trade ?

Any guidance is much appreciated. 

regards,
Patrick

**Comment by Lesigh-3100** [comment]
Hi Patrick,

Thanks for submitting this & for the detailed logs you provided.

FYI, error from Alchemy is actually automatically handled - when we get that error it updates the baseFee to be equal to the fee in the error message.

The error you see occurs when the transaction fails to build. This happens when there is a problem with the transaction - IE if the arbitrage opportunity was already closed by someone else, or if trades occurred on the pools your arb was routing through. 

I will make a note to improve the logging though, as it's not made clear. 

I'm closing this, but feel free to reopen if you think that it was a different issue that occurred. 

If you have any further questions regarding the bot, you can also ask in the Bancor Developers Telegram group: 
https://t.me/BancorDevelopers

Regards

**Reply by pblaas** [reply]
Thanks for your swift response and clarifying the issue.

**Comment by KimberlyBog** [comment]
interesting information

**Comment by dimonachi** [comment]
If you will fix it, write me please
